File: replace.js
const fs = require('fs');
const path = require('path');

const rootDir = 'C:\\Users\\Tx098\\Desktop\\presentacionTx\\sis257_NocturneTx';

const replacements = {
    "NOCTURNE: COLD STORAGE": "NOCTURNE: COLD STORAGE",
    "Nocturne: Cold Storage": "Nocturne: Cold Storage",
    "nocturnecoldstorage": "nocturnecoldstorage",
    "NOCTURNE_COLD_STORAGE": "NOCTURNE_COLD_STORAGE",
    "nocturne_cold_storage": "nocturne_cold_storage"
};

const ignoreDirs = new Set(["node_modules", "dist", ".git", "build"]);

function processFile(filepath) {
    try {
        let content = fs.readFileSync(filepath, 'utf8');
        let newContent = content;
        for (const [oldStr, newStr] of Object.entries(replacements)) {
            newContent = newContent.split(oldStr).join(newStr);
        }

        // Also replace the shield icon with the logo
        newContent = newContent.split('<img src="/logo.png" alt="Logo" class="animate-glow" style="height: 32px; width: auto;" />').join('<img src="/logo.png" alt="Logo" class="animate-glow" style="height: 32px; width: auto;" />');
        newContent = newContent.split('<img src="/logo.png" alt="Logo" style="height: 24px; width: auto;" />').join('<img src="/logo.png" alt="Logo" style="height: 24px; width: auto;" />');
        newContent = newContent.split('<img src="/logo.png" alt="Logo" class="me-2" style="height: 24px; width: auto;" />').join('<img src="/logo.png" alt="Logo" class="me-2" style="height: 24px; width: auto;" />');
        newContent = newContent.split('<img src="/logo.png" alt="Logo" style="height: 24px; width: auto;" />').join('<img src="/logo.png" alt="Logo" style="height: 24px; width: auto;" />');
        newContent = newContent.split('<img src="/logo.png" alt="Logo" class="mt-1" style="height: 24px; width: auto;" />').join('<img src="/logo.png" alt="Logo" class="mt-1" style="height: 24px; width: auto;" />');

        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
            console.log(`Updated: ${filepath}`);
        }
    } catch (e) {
        // Skip binary or unreadable files
    }
}

function walk(dir) {
    let list = fs.readdirSync(dir);
    for (let file of list) {
        let filepath = path.join(dir, file);
        let stat = fs.statSync(filepath);
        if (stat && stat.isDirectory()) {
            if (!ignoreDirs.has(file)) {
                walk(filepath);
            }
        } else {
            if (/\.(ts|tsx|vue|js|html|css|json|env|yml|md)$/i.test(file)) {
                processFile(filepath);
            }
        }
    }
}

walk(rootDir);

const sourceImg = 'C:\\Users\\Tx098\\.gemini\\antigravity\\brain\\180e57bf-1dee-4655-ac79-ff279f56e953\\owl_head_logo_1781641669628.png';
const destImg = path.join(rootDir, 'frontend_nocturne', 'public', 'logo.png');

if (fs.existsSync(sourceImg)) {
    fs.copyFileSync(sourceImg, destImg);
    console.log(`Copied logo to ${destImg}`);
} else {
    console.log(`Source image not found: ${sourceImg}`);
}

File: replace.py
import os
import shutil

# Paths
root_dir = r"C:\Users\Tx098\Desktop\presentacionTx\sis257_NocturneTx"

replacements = {
    "NOCTURNE: COLD STORAGE": "NOCTURNE: COLD STORAGE",
    "Nocturne: Cold Storage": "Nocturne: Cold Storage",
    "nocturnecoldstorage": "nocturnecoldstorage",
    "NOCTURNE_COLD_STORAGE": "NOCTURNE_COLD_STORAGE",
    "nocturne_cold_storage": "nocturne_cold_storage"
}

ignore_dirs = {"node_modules", "dist", ".git", "build"}

def process_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception as e:
        return # Skip binary files or unreadable files

    new_content = content
    for old, new in replacements.items():
        new_content = new_content.replace(old, new)
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated: {filepath}")

for root, dirs, files in os.walk(root_dir):
    dirs[:] = [d for d in dirs if d not in ignore_dirs]
    for file in files:
        if file.endswith((".ts", ".tsx", ".vue", ".js", ".html", ".css", ".json", ".env", ".yml", ".md")):
            process_file(os.path.join(root, file))

# Copy image
source_img = r"C:\Users\Tx098\.gemini\antigravity\brain\180e57bf-1dee-4655-ac79-ff279f56e953\owl_head_logo_1781641669628.png"
dest_img = os.path.join(root_dir, "frontend_nocturne", "public", "logo.png")

if os.path.exists(source_img):
    shutil.copy2(source_img, dest_img)
    print(f"Copied logo to {dest_img}")
else:
    print(f"Source image not found: {source_img}")

